tests/integration/src/test/scala/mesosphere/marathon/integration/SharedMemoryIntegrationTest.scala

```diff
@@ -0,0 +1,232 @@
+import mesosphere.marathon.integration.setup.{BaseMarathon, EmbeddedMarathonTest, MesosConfig}
+import mesosphere.marathon.raml.{Pod, Raml}
+import mesosphere.marathon.state.{HostVolume, VolumeMount}
+import mesosphere.{AkkaIntegrationTest, WhenEnvSet}
+import org.scalatest.time.{Seconds, Span}
+import play.api.libs.json.Json
+
+import scala.collection.immutable.Seq
+
+class SharedMemoryIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
+
+  val projectDir: String = sys.props.getOrElse("user.dir", ".")
+  override lazy val mesosConfig = MesosConfig(
+    launcher = "linux",
+    containerizers = "docker,mesos",
+    isolation = Some("filesystem/linux,docker/runtime,namespaces/ipc"),
+    imageProviders = Some("docker")
+  )
+
+  override implicit lazy val patienceConfig: PatienceConfig = PatienceConfig(timeout = Span(30, Seconds), interval = Span(1, Seconds))
+
+  "get correct shm size from pod" taggedAs WhenEnvSet(envVarRunMesosTests, default = "true") in {
+    Given("a pod with a single task and a volume")
+    val projectDir = sys.props.getOrElse("user.dir", ".")
+    val containerDir = "marathon"
+    val id = testBasePath / "simple-pod-with-shm-setup"
+
+    val shmSize = 11
+
+    def appMockCommand(port: String) =
+      s"""
+         |echo APP PROXY $$MESOS_TASK_ID RUNNING; \\
+         |$containerDir/python/app_mock.py $port $id v1 http://httpbin.org/anything
+        """.stripMargin
+
+    val pod = PodDefinition(
+      id = id,
+      containers = Seq(
+        MesosContainer(
+          name = "task1",
+          exec = Some(raml.MesosExec(raml.ShellCommand(appMockCommand("$ENDPOINT_TASK1")))),
+          resources = raml.Resources(cpus = 0.1, mem = 32.0),
+          endpoints = Seq(raml.Endpoint(name = "task1", hostPort = Some(0))),
+          healthCheck = Some(MesosHttpHealthCheck(portIndex = Some(PortReference("task1")), path = Some("/ping"))),
+          volumeMounts = Seq(
+            VolumeMount(Some("python"), s"$containerDir/python", true)
+          ),
+          linuxInfo = Some(state.LinuxInfo(
+            seccomp = None,
+            ipcInfo = Some(state.IPCInfo(
+              ipcMode = state.IpcMode.Private,
+              shmSize = Some(shmSize)
+            ))
+          ))
+        )
+      ),
+      volumes = Seq(
+        HostVolume(Some("python"), s"$projectDir/src/test/resources/python")
+      ),
+      unreachableStrategy = state.UnreachableDisabled,
+      upgradeStrategy = state.UpgradeStrategy(0.0, 0.0),
+      networks = Seq(HostNetwork),
+      instances = 1,
+      role = BaseMarathon.defaultRole
+    )
+
+    When("The pod is deployed")
+    val createResult = marathon.createPodV2(pod)
+    createResult should be(Created)
+    waitForDeployment(createResult)
+    val ipcInfo:String = eventually {
+      marathon.status(pod.id) should be(Stable)
+      val status = marathon.status(pod.id).value
+      val hosts = status.instances.flatMap(_.agentHostname)
```

Comment:
Let's use inside, instead

```suggestion
      val host = inside(status.instances.flatMap(_.agentHostname)) {
        case Seq(host) => host
      }
```

This will avoid as having to call .head later on

Reply:
You just need to mix in the Inside trait provided by Scala test into this test suite.

Reply:
Good idea, done